tests/ui/js/storyTellingServerUITests.js

```diff
@@ -0,0 +1,213 @@
+            theme: "learningReflections",
+            baseTheme: "base",
+            authoringEnabled: true
+        }
+    };
+
+    var browsePageOptions = {
+        components: {
+            uio: {
+                options: {
+                    terms: {
+                        "templatePrefix": "../../node_modules/infusion/src/framework/preferences/html",
+                        "messagePrefix": "../../messages/uio"
+                    },
+                    "tocTemplate": "../../node_modules/infusion/src/components/tableOfContents/html/TableOfContents.html"
+                }
+            },
+            menu: {
+                container: "#testMenu"
+            },
+            storyBrowser: {
+                container: "#testStoryBrowser"
+            }
+        }
+    };
+
+    fluid.defaults("sjrk.storyTelling.storyTellingServerUiTester", {
+        gradeNames: ["fluid.test.testCaseHolder"],
+        modules: [{
+            name: "Test Storytelling Server UI code",
+            tests: [{
+                name: "Test themed page loading functions",
+                expect: 8,
+                sequence: [{
+                    // call the load themed page function, forcing the base theme response
+                    task: "sjrk.storyTelling.storyTellingServerUiTester.loadThemedPageSingleTest",
+                    args: ["sjrk.storyTelling.testUtils.callbackVerificationFunction", themedPageTestCases.base.theme],
+                    resolve: "jqUnit.assertDeepEq",
+                    resolveArgs: ["The themed page load resolved as expected", themedPageTestCases.base, "{arguments}.0"]
+                },{
+                    // call the load themed page function, forcing the Learning Reflections theme response
+                    task: "sjrk.storyTelling.storyTellingServerUiTester.loadThemedPageSingleTest",
+                    args: ["sjrk.storyTelling.testUtils.callbackVerificationFunction", themedPageTestCases.learningReflections.theme],
+                    resolve: "jqUnit.assertDeepEq",
+                    resolveArgs: ["The themed page load resolved as expected", themedPageTestCases.learningReflections, "{arguments}.0"]
+                },{
+                    funcName: "sjrk.storyTelling.storyTellingServerUiTester.assertCustomCssLoaded",
+                    args: ["learningReflections.css", 1]
+                },{
+                    // test the CSS/JS injection function directly
+                    funcName: "sjrk.storyTelling.loadCustomThemeFiles",
+                    args: ["sjrk.storyTelling.testUtils.callbackVerificationFunction", {"theme": "learningReflections"}]
+                },{
+                    funcName: "sjrk.storyTelling.storyTellingServerUiTester.assertCustomCssLoaded",
+                    args: ["learningReflections.css", 2]
+                },{
+                    // load a base Browse page and check that the stories are present
+                    task: "sjrk.storyTelling.loadBrowse",
```

Comment:
Any time there's a call to loadBrowse we need to arrange for cleanup - I suggest packaging the assertion functions into a reusable utility that as well as making the assertion calls storyBrowser.destroy() once it is done

Reply:
A slightly cleaner structure would just have one testEnvironment per use of a "page" and then repackage sjrk.storyTelling.loadThemedPage rather than as a standalone function, a definition for a "createOnEvent" subcomponent of a PageLoader component which fires on the completion of the I/O for /stories (and /clientConfig, with boiling, etc.). Then you could rely on the standard teardown of the testEnvironment to arrange for destruction of the PageLoader once you were finished tinkering with it in the fixtures. This might be enough in itself to stop the race on constructing the 2nd and further UIO components into the same page but I will continue to investigate

Reply:
I have started this work but have found it to be rather complex and outside of the scope of SJRK-255. I started implementing a `pageLoader` grade that used a `fluid.dataSource.URL` to load in the page data, but I ran into a roadblock while attempting to make the `url` option of the DataSource configurable.

This roadblock has been captured in FLUID-6449:
https://issues.fluidproject.org/browse/FLUID-6449

I have also filed a fresh Jira that copies these review comments and the one above them:
https://issues.fluidproject.org/browse/SJRK-320

and my preliminary work on this can be found in this branch:
https://github.com/BlueSlug/sjrk-story-telling/tree/SJRK-320